sensor/common/compliance/nodescan_handler.go

```diff
@@ -4,21 +4,24 @@ import (
 	"github.com/stackrox/rox/generated/internalapi/central"
 	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/sync"
 	"github.com/stackrox/rox/sensor/common"
 )
 
 // NodeScanHandler is responsible for handling the arriving NodeScanV2 messages, processing then, and sending them to central
 type NodeScanHandler interface {
 	common.SensorComponent
+	Stopped() concurrency.ReadOnlyErrorSignal
 }
 
 // NewNodeScanHandler returns a new instance of a NodeScanHandler
 func NewNodeScanHandler(ch <-chan *storage.NodeScanV2) NodeScanHandler {
 	return &nodeScanHandlerImpl{
-		nodeScans: ch,
-		toCentral: make(chan *central.MsgFromSensor),
-
-		stopC:    concurrency.NewErrorSignal(),
-		stoppedC: concurrency.NewErrorSignal(),
+		nodeScans:    ch,
+		toCentral:    make(chan *central.MsgFromSensor),
```

Comment:
There's no need to initialize `toCentral` if `nodeScanHandlerImpl.run()` would anyway create a new one.

Reply:
I didn't follow up on this. Last time I moved the initialization outside of the constructor I got race detection. Will check it again.

Reply:
I set that to nil, however it must remain protected with the mutex to avoid race when running tests.